torch/csrc/jit/script/compiler.cpp

```diff
@@ -2532,22 +2604,35 @@ std::vector<Argument> parseArgsFromDecl(Decl decl, bool is_method) {
 
     TypePtr type;
     c10::optional<int32_t> N;
+    c10::optional<IValue> default_value;
+    bool is_template = false;
 
     //BroadcastList list can only appear at the argument level
     if (auto maybe_broad_list = handleBroadcastList(decl_arg.type())) {
       type = maybe_broad_list->first;
       N = maybe_broad_list->second;
+    } else if (auto maybe_template = handleTemplate(decl_arg.type())) {
```

Comment:
In its current form I believe this will accept `a : Template[Bool, True]  = 3` `True` will be forgotten and there will be no check that `3` is a valid default value for the type.